documentation/config_file_auth_migration.md

```diff
@@ -0,0 +1,267 @@
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ...
+    ```
+
+Our HOCON `auth.conf` file should now allow all clients to GET and POST to the `/puppet/v3/environments` endpoint, and should look like this:
+
+``` hocon
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ]
+}
+```
+
+### Converting more complex rules
+
+#### Paths set by regular expressions
+
+To convert a regular expression path, enclose it in double quotation marks and solidus characters (`/`), and set the `type` to regex.
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
+
+Deprecated:
+
+```
+path ~ ^/puppet/v3/catalog/([^/]+)$
+```
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            path: "/^\/puppet\/v3\/catalog\/([^\/]+)$/"
+            type: regex
+...
+```
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
+
+Backreferencing works the same way it does in the deprecated format.
+
+Deprecated:
+
+```
+path ~ ^/puppet/v3/catalog/([^/]+)$
+allow $1
+```
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            path: "/^\/puppet\/v3\/catalog\/([^\/]+)$/"
+            type: regex
+        }
+        allow: $1
+...
+```
+
+#### Multiple `method` indirectors
+
+If a deprecated rule has multiple `method` indirectors, combine all of the related HTTP methods to the HOCON `method` array.
+
+Deprecated:
+
+```
+method find, save
+```
+
+The deprecated find indirector corresponds to the GET and POST methods, and the save indirector corresponds to the PUT method. In the HOCON format, simply combine these methods in an array.
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            ...
+            method: [get, post, put]
+        }
+```
+
+#### Search indirector for `method`
+
+There's no direct equivalent to the search indirector for the deprecated `method` setting. Create the equivalent rule by passing GET and POST to `method` and specifying endpoint paths using the `path` parameter.
+
+Deprecated:
+
+```
+
+``` hocon
+...
+        match-request: {
+            path: "/(s|_search)$/"
```

Comment:
Same comment here as above about not needing the leading and trailing solidus characters.  Might be good for this one to include an example of a full endpoint that has a plural endpoint, like:

```
path: "^/puppet/v3/file_metadatas?/user_files/"
```